mempool/src/shared_mempool/network.rs

```diff
@@ -425,40 +420,104 @@ impl<NetworkClient: NetworkClientInterface<MempoolSyncMsg>> MempoolNetworkInterf
-                Some(id) => {
-                    let metric_label = if Some(id) == expired_batch_id {
+        let (message_id, transactions, metric_label) =
+            match std::cmp::max(expired_message_id, retry_message_id) {
+                Some(message_id) => {
+                    let metric_label = if Some(message_id) == expired_message_id {
                         Some(counters::EXPIRED_BROADCAST_LABEL)
                     } else {
                         Some(counters::RETRY_BROADCAST_LABEL)
                     };
 
-                    let txns = mempool.timeline_range(&id.0);
-                    (id.clone(), txns, metric_label)
+                    let txns = message_id
+                        .decode()
+                        .into_iter()
+                        .flat_map(|(sender_bucket, start_end_pairs)| {
+                            self.prioritized_peers_state
+                                .get_sender_bucket_priority_for_peer(&peer, sender_bucket)
+                                .map_or_else(Vec::new, |priority| {
+                                    mempool
+                                        .timeline_range(sender_bucket, start_end_pairs)
+                                        .into_iter()
+                                        .map(|(txn, ready_time)| {
+                                            (txn, ready_time, priority.clone())
+                                        })
+                                        .collect::<Vec<_>>()
+                                })
+                        })
+                        .collect::<Vec<_>>();
+                    (message_id.clone(), txns, metric_label)
                 },
                 None => {
                     // Fresh broadcast
-                    let before = match peer_priority {
-                        BroadcastPeerPriority::Primary => None,
-                        BroadcastPeerPriority::Failover => Some(
-                            Instant::now()
-                                - Duration::from_millis(
-                                    self.mempool_config.shared_mempool_failover_delay_ms,
-                                ),
-                        ),
-                    };
-                    let (txns, new_timeline_id) = mempool.read_timeline(
-                        &state.timeline_id,
-                        self.mempool_config.shared_mempool_batch_size,
-                        before,
-                        peer_priority,
-                    );
+
+                    // If the peer doesn't have any sender_buckets assigned, let's not broadcast to the peer
+                    let mut sender_buckets: Vec<(MempoolSenderBucket, BroadcastPeerPriority)> =
+                        self.prioritized_peers_state
+                            .get_sender_buckets_for_peer(&peer)
+                            .ok_or_else(|| {
+                                info!(
```

Comment:
This log is way too spammy. Probably just remove (it seems too spammy even for debug)

Reply:
Done.